lsls/lsls.c

```diff
@@ -1,18 +1,44 @@
 #include <stdio.h>
 #include <dirent.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <string.h>
+#include <stdlib.h>
 
 /**
  * Main
  */
 int main(int argc, char **argv)
 {
   // Parse command line
+  char *directory = argv[1] ? argv[1] : ".";
 
-  // Open directory
+  //Open directory
+  DIR *d = opendir(directory);
 
   // Repeatly read and print entries
+  struct dirent *entry;
+  while ((entry = readdir(d)) != NULL)
+  {
+    char *name = entry->d_name;
+    char path[strlen(directory) + strlen(name) + 1];
+
+    strcpy(path, directory);
+    strcat(path, "/");
+    strcat(path, name);
+
+    // I don't know how this black magic works.
+    // sprintf(path, "%s/%s", directory, name);
+
+    struct stat buf;
+    stat(path, &buf);
+
+    // printf("%s:\n", path);
+    printf("%10ld %s\n", buf.st_size, name);
```

Comment:
You're missing an "l" in `%10lld`, buf.st_size is of type "long long" as opposed to "long" which is a type that longs fewer numbers. You should notice an error in your console when you run `make`